Detectors/DCS/testWorkflow/DCSDataProcessorSpec.h

```diff
@@ -0,0 +1,244 @@
+
+    // delta map
+    const auto* dhDelta = o2::header::get<o2::header::DataHeader*>(pc.inputs().get("inputDelta").header);
+    auto szDelta = dhDelta->payloadSize;
+    int nDPsDelta = szDelta / sizeof(DPCOM);
+    LOG(INFO) << "Number of DPs received (delta map) = " << nDPsDelta;
+    std::unordered_map<DPID, DPVAL> dcsmapDelta;
+    LOG(INFO) << "TF: " << tfid << " -->  building (delta) unordered_map...";
+    s.Start();
+    for (int i = 0; i < nDPsDelta; i++) {
+      memcpy(&dptmp, rawcharDelta + i * sizeof(DPCOM), sizeof(DPCOM));
+      dcsmapDelta[dptmp.id] = dptmp.data;
+      LOG(DEBUG) << "Reading from generator: i = " << i << ", DPCOM = " << dptmp;
+      LOG(DEBUG) << "Reading from generator: i = " << i << ", DPID = " << dptmp.id;
+    }
+    s.Stop();
+    LOG(INFO) << "TF: " << tfid << " -->  ...unordered_map (delta) built = " << s.RealTime() << ", cpuTime = " << s.CpuTime();
+
+    if (tfid % 6000 == 0) {
+      //for (int idet = 0; idet < detVect.size(); idet++) {
+      for (int idet = 0; idet < 1; idet++) { // for now I test only 1 DCS processor
+        LOG(INFO) << "TF: " << tfid << " -->  starting processing...";
+        s.Start();
+        mDCSprocVect[idet].processMap(dcsmap, false);
+        s.Stop();
+        LOG(INFO) << "TF: " << tfid << " -->  ...processing done: realTime = " << s.RealTime() << ", cpuTime = " << s.CpuTime();
+      }
+    } else {
+      if (mProcessFullDeltaMap) {
+        //for (int idet = 0; idet < detVect.size(); idet++) {
+        for (int idet = 0; idet < 1; idet++) { // for now I test only 1 DCS processor
+          LOG(INFO) << "TF: " << tfid << " -->  starting (delta) processing...";
+          s.Start();
+          mDCSprocVect[idet].processMap(dcsmapDelta, true);
+          s.Stop();
+          LOG(INFO) << "TF: " << tfid << " -->  ...processing (delta) done: realTime = " << s.RealTime() << ", cpuTime = " << s.CpuTime();
+        }
+      } else {
+
+        // processing per DP found in the map, to be done in case of a delta map
+
+        bool resetStopwatch = true;
+        LOG(INFO) << "TF: " << tfid << " -->  starting (delta) processing for detector...";
+        for (const auto& dpcom : dcsmapDelta) {
+          std::vector<int> detVect = mDetectorAlias[dpcom.first];
+          //for (int idet = 0; idet < detVect.size(); idet++) {
+          for (int idet = 0; idet < 1; idet++) { // for now I test only 1 DCS processor
+            s.Start(resetStopwatch);
+            mDCSprocVect[idet].processDP(dpcom);
+            s.Stop();
+            resetStopwatch = false;
+          }
+        }
+        LOG(INFO) << "TF: " << tfid << " -->  ...processing (delta) in detector loop done: realTime = " << s.RealTime() << ", cpuTime = " << s.CpuTime();
+        // now preparing CCDB object
+        //for (int idet = 0; idet < detVect.size(); idet++) {
+        for (int idet = 0; idet < 1; idet++) { // for now I test only 1 DCS processor
```

Comment:
why the ccdb object preparation is called only in per-DP loop?

Reply:
In case we pass the full map, the CCDB object is created in the processMap method directly.

Reply:
I see, ok. I hope in real life we will not call ``prepareCCDBobject`` for every TF / map update. But as a demo it is fine.

Reply:
I hope so too :-) As you said, it is a demo.